homeassistant/components/nsw_fuel_station/sensor.py

```diff
@@ -181,6 +122,18 @@ def unit_of_measurement(self) -> str:
         """Return the units of measurement."""
         return f"{CURRENCY_CENT}/{VOLUME_LITERS}"
 
-    def update(self):
-        """Update current conditions."""
-        self._station_data.update()
+    def _get_station_name(self):
+        default_name = f"station {self._station_id}"
+        if self.coordinator.data is None:
+            return default_name
+
+        station = self.coordinator.data.stations.get(self._station_id)
+        if station is None:
+            return default_name
+
+        return station.name
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique ID."""
+        return f"nsw_fuel_station_{self._station_id}_{self._fuel_type}"
```

Comment:
Ids are unique to the integration and platform so there is no need to prefix it with nsw_fuel_station

Reply:
Today I learned! Thanks